python/packages/isce3/math/offsets_polyfit.py

```diff
@@ -0,0 +1,508 @@
+    A : ndarray of shape (N, M)
+        The design matrix, where `N` is the number of observations and
+        `M` is the number of polynomial coefficients, given by:
+
+        M = (degree + 1) * (degree + 2) / 2
+    """
+    N = lines_n.size
+    A = np.zeros((N, ncoeffs(degree)), dtype=float)
+    idx = 0
+    for j in range(degree + 1):
+        for k in range(j + 1):
+            A[:, idx] = (lines_n ** (j - k)) * (pixels_n ** k)
+            idx += 1
+    return A
+
+
+def cholesky_solve(N_mat, rhs):
+    """
+    Solve a symmetric positive-definite linear system using Cholesky decomposition.
+
+    This function solves the system:
+
+        N_mat * x = rhs
+
+    where `N_mat` is a symmetric positive-definite matrix, by performing
+    a Cholesky decomposition `N_mat = L @ L.T` and solving the two
+    triangular systems `L y = rhs` and `L.T x = y`.
+
+    Parameters
+    ----------
+    N_mat : (M, M) array_like
+        Symmetric positive-definite coefficient matrix.
+    rhs : (M,) or (M, K) array_like
+        Right-hand side vector or matrix.
+
+    Returns
+    -------
+    x : ndarray
+        Solution vector or matrix satisfying `N_mat @ x = rhs`.
+    L : ndarray
+        Lower-triangular Cholesky factor of `N_mat`.
+    """
+
+    L = np.linalg.cholesky(N_mat)
+    y = np.linalg.solve(L, rhs)
+    x = np.linalg.solve(L.T, y)
+
+    return x, L
+
+def invert_from_cholesky(L):
+    """
+    Compute the inverse of a symmetric positive-definite matrix from its
+    Cholesky factorization.
+
+    Given a matrix `A` such that `A = L @ L.T`, this function computes the
+    inverse of `A` as:
+
+        A_inv = (L^-1).T @ (L^-1)
+
+    Parameters
+    ----------
+    L : (M, M) array_like
+        Lower-triangular Cholesky factor of a symmetric positive-definite
+        matrix `A`.
+
+    Returns
+    -------
+    A_inv : (M, M) ndarray
+        Inverse of the original matrix `A`.
+    """
+
+    Linv = np.linalg.inv(L)
+    return Linv.T @ Linv
+
+def polyfit_offsets(
+    data,degree=2,
+    crit_value=0.1,
+    max_iterations=50,
+    minL=None, maxL=None,
+    minP=None, maxP=None,
+    prf=None, abw=None,
+    rsr=None, rbw=None):
+
+    """
+    Robustly fit 2D polynomial models to SAR line/pixel offsets with iterative
+    outlier rejection.
+
+    The function estimates two polynomial surfaces (line and pixel directions)
+    from tie-point measurements by building a 2D polynomial design matrix up to
+    `degree`, solving with weighted normal equations, and iteratively removing
+    outliers using a w-test controlled by `crit_value`.
+
+    Parameters
+    ----------
+    data : (N, 6) array_like
+        Input tie-point array where each row is:
+        ``[id, line, pixel, dL, dP, corr_peak]``.
+        - *id* : int-like identifier.
+        - *line, pixel* : coordinates of the tie point.
+        - *dL, dP* : measured offsets in line and pixel directions.
+        - *corr_peak* : correlation peak or quality metric.
```

Comment:
```suggestion
        - *corr_peak* : correlation peak or quality metric in interval (0, 1].
        Measurements with larger values get more weight.
```
A bigger number means the measurement should get more weight, right? That seems to be the behavior below. It's not necessarily intuitive since often one would supply measurement variances (smaller is better).

Reply:
You are right, the bigger number means more weights, and has changed the docstring to make it clear